--- Tk-Zinc-3.306/PostScript.c.orig	2020-11-25 20:23:17.000000000 +0100
+++ Tk-Zinc-3.306/PostScript.c	2025-09-28 13:36:02.724428041 +0200
@@ -305,7 +305,7 @@ ZnPostScriptCmd(ZnWInfo        *wi,
   ps_info.prolog = 1;
   Tcl_InitHashTable(&ps_info.fontTable, TCL_STRING_KEYS);
   result = Tk_ConfigureWidget(wi->interp, wi->win, config_specs,
-                              argc-2, (CONST char **) argv+2,
+                              argc-2, argv+2,
                               (char *) &ps_info,
                               TK_CONFIG_ARGV_ONLY|TK_CONFIG_OBJS);
   if (result != TCL_OK) {
--- Tk-Zinc-3.306/tkZinc.c.orig	2020-11-25 20:23:17.000000000 +0100
+++ Tk-Zinc-3.306/tkZinc.c	2025-09-28 13:46:10.401208797 +0200
@@ -8999,11 +8999,11 @@ InitZinc(Tcl_Interp *interp) {
   ZnTesselator.tess = gluNewTess();
   ZnTesselator.combine_list = NULL;
         ZnTesselator.combine_length = 0;
-  gluTessCallback(ZnTesselator.tess, GLU_TESS_BEGIN_DATA, ZnTessBegin);
-  gluTessCallback(ZnTesselator.tess, GLU_TESS_VERTEX_DATA, ZnTessVertex);
-  gluTessCallback(ZnTesselator.tess, GLU_TESS_END_DATA, ZnTessEnd);
-  gluTessCallback(ZnTesselator.tess, GLU_TESS_COMBINE_DATA, ZnTessCombine);
-  gluTessCallback(ZnTesselator.tess, GLU_TESS_ERROR_DATA, ZnTessError);
+  gluTessCallback(ZnTesselator.tess, GLU_TESS_BEGIN_DATA, (_GLUfuncptr) ZnTessBegin);
+  gluTessCallback(ZnTesselator.tess, GLU_TESS_VERTEX_DATA, (_GLUfuncptr) ZnTessVertex);
+  gluTessCallback(ZnTesselator.tess, GLU_TESS_END_DATA, (_GLUfuncptr) ZnTessEnd);
+  gluTessCallback(ZnTesselator.tess, GLU_TESS_COMBINE_DATA, (_GLUfuncptr) ZnTessCombine);
+  gluTessCallback(ZnTesselator.tess, GLU_TESS_ERROR_DATA, (_GLUfuncptr) ZnTessError);
   gluTessNormal(ZnTesselator.tess, 0.0, 0.0, -1.0);
 
   /*
--- Tk-Zinc-3.306/Field.c.orig	2020-11-25 20:23:17.000000000 +0100
+++ Tk-Zinc-3.306/Field.c	2026-08-18 00:07:33.959494963 +0200
@@ -1617,7 +1617,7 @@
  */
 static void
 FieldsEngine(ZnFieldSet field_set,
-             void       (*cb)())
+             void       (*cb)(ZnWInfo*,Field,ZnBBox*,ZnBBox*,ZnPoint*,ZnBBox*,int,int, int))
 {
   ZnWInfo       *wi = field_set->item->wi;
   /*int         i;      This one *NEED* to be an int */
--- Tk-Zinc-3.306/OverlapMan.c.orig	2020-11-25 20:23:17.000000000 +0100
+++ Tk-Zinc-3.306/OverlapMan.c	2026-08-18 00:07:52.492728829 +0200
@@ -98,8 +98,8 @@
 
 typedef struct _ZINCS {
   void  *rw;
-  void  *(*_next_track)();
-  void  (*_set_label_angle)();
+  void  *(*_next_track)(void *, void *, int *, int *, int *, int *, int *, int *, int *, int *, int *, int *);
+  void  (*_set_label_angle)(void *, void *, int, int);
   INFOS *infos;
   int   NBinfos;
   int   NBalloc_infos;
@@ -500,13 +500,13 @@
     }
     
     info1.New_Track = TRUE;
-    SetTrackInitValues(iw, pos);
+    SetTrackInitValues();
 
     wr[iw].NBinfos++;
   }
   else {
     if (info1.visibility == FALSE) {  
-      SetTrackInitValues( iw, pos);
+      SetTrackInitValues();
     }
     else {
       SetTrackCurrentValues(iw, pos);
